### PR TITLE
Closes #272, #253, #72 continues deprecation process and adds some doc updates

### LINKED ISSUE #272: Remove deprecated functions from admiraldev

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiraldev/devel/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -14,12 +14,21 @@ expressions (#241)
 ## Breaking Changes
 
 - `assert_order_vars()` was deprecated in favor of `assert_expr_list()`. (#241)
+- The following functions have been deprecated from previous admiral versions using the next phase of the deprecation process: (#272)
+
+  - `quo_c()`
+  - `quo_not_missing()`
+  - `replace_values_by_names()`
+  - `replace_symbol_in_quo()`
+
```
Comment (bms63) [inline NEWS.md:23]: were some of these removed completely - can we list them out separately from those that are producing warnings please - jut so we can keep track

Comment (zdz2101) [reply NEWS.md:23]: Removal would be the next step, these were the ones in phase 1 with `deprecate_warn` and I've swapped in the `deprecate_stop`

File: tests/testthat/test-quo.R [hunk 2]
```diff
@@ -134,9 +135,9 @@ test_that("replace_values_by_names Test 11: names of argument is NULL", {
 
 ## Test 12: warning if quosures argument is used ----
 test_that("replace_values_by_names Test 12: warning if quosures argument is used", {
-  expect_warning(
+  expect_error(
     replace_values_by_names(quosures = rlang::quos(STUDYID, USUBJID)),
-    class = "lifecycle_warning_deprecated"
+    class = "lifecycle_error_deprecated"
   )
 })
```
Comment (bms63) [inline tests/testthat/test-quo.R:142]: can you check that these don't admit error/warnings when installing, e.g. #268 .  I think we can suppress them??

Comment (zdz2101) [reply tests/testthat/test-quo.R:142]: Looks resolved at least on my end!

Comment (bms63) [reply tests/testthat/test-quo.R:142]: fabulous

File: NEWS.md (hunk 1 above)
Comment (bms63) [reply NEWS.md:23]: okay cool thank you for checking.

Comment (bms63) [reply NEWS.md:23]: can you make a new issue for Q3 release (you can even make the new label! ) that says these four functions should be removed for Q3 release.  Just so we don't lose track

File: R/quo.R [hunk 3]
```diff
@@ -9,12 +9,12 @@
...
-#' @keywords quo
-#' @family quo
+#' @keywords deprecated
+#' @family deprecated
 #'
 #' @export
 quo_c <- function(...) {
-  deprecate_warn(
+  deprecate_stop(
     "0.10.0",
```
Comment (zdz2101) [inline R/quo.R:18]: The one thing I think we should course-correct right now is when we deprecate on `admiraldev` it should reflect the versions of `admiraldev` or `admiral` ? Here its `admiral`

File: R/quo.R [hunk 4]
```diff
@@ -64,12 +60,12 @@ expr_c <- function(...) {
...
-#' @keywords quo
-#' @family quo
+#' @keywords deprecated
+#' @family deprecated
 #'
 #' @export
 quo_not_missing <- function(x) {
-  deprecate_warn(
+  deprecate_stop(
     "0.3.0",
```
Comment (zdz2101) [inline R/quo.R:69]: and here its `admiraldev` @bms63

File: R/quo.R (hunk 3 above)
Comment (bms63) [reply R/quo.R:18]: oohhh...this gets interesting.  I think it should be `admiraldev`. We use them in admiral, but if we are using them then the message should be telling us to switch to a function in `admiraldev`!

Comment (zdz2101) [reply R/quo.R:18]: image

Now that you mention, the default lifecycle messaging actually makes it quite clear its `admiraldev` so I'll swap out all these "0.10.0" for "0.3.0" when the deprecation phase 1 process started

File: NEWS.md (hunk 1 above)
Comment (bundfussr) [reply NEWS.md:23]: `replace_values_by_names()` was not deprecated but the `quosures` argument was replaced by the `expressions` argument.

File: R/quo.R [hunk 5]
```diff
@@ -97,8 +84,8 @@ quo_not_missing <- function(x) {
 #' @param quosures *Deprecated*, please use `expressions` instead.
 #'
 #'
-#' @keywords quo
-#' @family quo
+#' @keywords deprecated
+#' @family deprecated
```
Comment (bundfussr) [inline R/quo.R:88]: Should not be changed as the function is not deprecated but only one of its arguments.

**Review by bundfussr — CHANGES_REQUESTED** [review]

**Review by bms63 — APPROVED** [review]
LGTM!

**Comment by zdz2101** [comment]
@bundfussr can you re-take a look at this?

**Review by bundfussr — APPROVED** [review]
